packages/core/src/sso/SamlSsoConnector/index.ts

```diff
@@ -0,0 +1,38 @@
+import { ConnectorError, ConnectorErrorCodes } from '@logto/connector-kit';
+import { type SsoConnector } from '@logto/schemas';
+
+import SamlConnector from '../SamlConnector/index.js';
+import { type SingleSignOnFactory } from '../index.js';
+import { type SingleSignOn, SsoProviderName } from '../types/index.js';
+import { baseSamlConnectorConfigGuard } from '../types/saml.js';
+
+export class SamlSsoConnector extends SamlConnector implements SingleSignOn {
+  constructor(
+    private readonly _data: SsoConnector,
+    tenantId: string
+  ) {
+    const parseConfigResult = baseSamlConnectorConfigGuard.safeParse(_data.config);
+
+    if (!parseConfigResult.success) {
+      throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, parseConfigResult.error);
+    }
+
+    super(parseConfigResult.data, tenantId, _data.id);
+  }
```

Comment:
For SAML connector this part could be a little bit complicate.  Depends on the input guard, we need to further verify the config:

if(metadataUrl) return;

if (metadata) {
 //decode and parse the output
}

parse the reset and strictly guard the mandatory fields.